clang/include/clang/Basic/Attr.td

```diff
@@ -1023,6 +1023,26 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def SYCLScope : Attr {
+  let Spellings = [GNU<"sycl_scope">];
+  let Args = [EnumArgument<"level", "Level", ["work_group", "work_item"],
+                                             ["WorkGroup", "WorkItem"]>];
+  let Subjects = SubjectList<[Function]>;
+  let LangOpts = [SYCLIsDevice];
+
+  let AdditionalMembers = [{
+    bool isWorkGroup() const {
+      return getLevel() == SYCLScopeAttr::WorkGroup;
+    }
+
+    bool isWorkItem() const {
+      return getLevel() == SYCLScopeAttr::WorkItem;
+    }
+  }];
+
+  let Documentation = [Undocumented];
```

Comment:
Unless we document this, it'll face a massive up hill battle getting into community Clang.

Reply:
Since I'm removing the spelling, this should not be a problem, correct? Or should I still add the doc?

Reply:
It might be useful for the reviewer to know what they are reading... :-)